airflow/providers/amazon/aws/operators/step_function.py

```diff
@@ -74,9 +88,25 @@ def execute(self, context: Context):
             raise AirflowException(f"Failed to start State Machine execution for: {self.state_machine_arn}")
 
         self.log.info("Started State Machine execution for %s: %s", self.state_machine_arn, execution_arn)
-
+        if self.deferrable:
+            self.defer(
+                trigger=StepFunctionsStartExecutionTrigger(
+                    execution_arn=execution_arn,
+                    waiter_delay=self.waiter_delay,
+                    waiter_max_attempts=self.waiter_max_attempts,
+                    aws_conn_id=self.aws_conn_id,
+                    region_name=self.region_name,
+                ),
+                method_name="execute_complete",
+                timeout=timedelta(seconds=self.waiter_max_attempts * self.waiter_delay),
```

Comment:
no extra time buffer here ? I seem to recall we gave it 30 extra seconds in other operators ?

Reply:
We were initially giving an extra 60 seconds along with an explanation for why we were doing that. I've personally stopped doing that after #31757 , which adds a log message indicating a timeout.

Reply:
ok, makes sense.
I also wonder if we really care about those timeouts since by design we shouldn't run longer than that. It's "belt and suspenders".

Reply:
The timeout was added from a discussion where it was pointed out that Triggers could potentially die and be respawned, which would cause the timer to reset if it was stored inside the Trigger itself. This way, the timeout is separate from the Trigger, so if something happens to the Trigger, the timeout doesn't reset.